lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb

```diff
@@ -103,9 +102,15 @@ def generate_settings_to_import_pod_targets
             end
             build_settings = {
               'PODS_FRAMEWORK_BUILD_PATH' => target.scoped_configuration_build_dir,
-              # Make headers discoverable by `import "…"`
-              'OTHER_CFLAGS' => '$(inherited) ' + XCConfigHelper.quote(header_search_paths, '-iquote'),
+              # Make framework headers discoverable by `import "…"`
+              'OTHER_CFLAGS' => '$(inherited) ' + XCConfigHelper.quote(framework_header_search_paths, '-iquote'),
             }
+            if pod_targets.any? { |t| !t.should_build? }
+              # Make library headers discoverale by `#import "…"`
+              library_header_search_paths = target.sandbox.public_headers.search_paths(target.platform)
```

Comment:
I think so, but I think it would involve modifying how the headers are linked in the sandbox. I don't see a clean way to get at the sandboxed public headers per pod – they're all set at the same time, and exposed through ```sandbox.public_headers```. The method that does this, ```FileReferencesInstaller::link_headers``` doesn't currently check if a pod is being integrated statically or dynamically, and so headers for pods that are integrated as frameworks are still linked into the public headers.

I considered trying to fix that in this pull request, but it seemed more appropriate as a separate PR. If there's a way to get at the sandboxed headers individually, I'll gladly do that in this PR, but I didn't see it.

https://github.com/CocoaPods/CocoaPods/blob/cc831ac008672e3633f0e2aa62b94f1e51d8e530/lib/cocoapods/installer/file_references_installer.rb#L114